Sources/Core/FileGenerator.swift

```diff
@@ -21,10 +21,253 @@ public enum GenerationParameterType {
     case packageName
 }
 
+// Most of these are derived from https://www.binpress.com/tutorial/objective-c-reserved-keywords/43
+// other language conflicts should be ideally added here.
+// TODO: Find a way to separate this by language since the reserved keywords will differ.
+let objectiveCReservedWordReplacements = [
+    "description": "description_text",
+    "id": "identifier",
+]
+
+let objectiveCReservedWords = Set<String>([
+    "@catch()",
+    "@class",
+    "@dynamic",
+    "@end",
+    "@finally",
+    "@implementation",
+    "@interface",
+    "@private",
+    "@property",
+    "@protected",
+    "@protocol",
+    "@public",
+    "@selector",
+    "@synthesize",
+    "@throw",
+    "@try",
+    "BOOL",
+    "Class",
+    "IMP",
+    "NO",
+    "NULL",
+    "Protocol",
+    "SEL",
+    "YES",
+    "_Bool",
+    "_Complex",
+    "_Imaginery",
+    "atomic",
+    "auto",
+    "break",
+    "bycopy",
+    "byref",
+    "case",
+    "char",
+    "const",
+    "continue",
+    "default",
+    "do",
+    "double",
+    "else",
+    "enum",
+    "extern",
+    "float",
+    "for",
+    "goto",
+    "id",
+    "if",
+    "in",
+    "inline",
+    "inout",
+    "int",
+    "long",
+    "nil",
+    "nonatomic",
+    "oneway",
+    "out",
+    "register",
+    "restrict",
+    "retain",
+    "return",
+    "self",
+    "short",
+    "signed",
+    "sizeof",
+    "static",
+    "struct",
+    "super",
+    "switch",
+    "typedef",
+    "union",
+    "unsigned",
+    "void",
+    "volatile",
+    "while",
+])
+
+// TODO: "id" is technically allowed. It's possible not everyone wants this replacement.
```

Comment:
Can you file an issue to revisit this? I don't want to necessarily block internal progress but I think this warrants a further discussion

Reply:
https://github.com/pinterest/plank/issues/189